File: sort/selection_sort.py
def selection_sort(arr):
    n = len(arr)

    for i in range(n-1):
        min = i

        for curr_idx in range(i+1, n):
            if arr[min] > arr[curr_idx]:
                min = curr_idx

        if min != i:
            temp = arr[i]
            arr[i] = arr[min]
            arr[min] = temp
    return arr


def main():
    list = [99, 0, 5, 20, 123, 0, -1, 72, 21, 22, 13, 8, 7, 67, 29, 1, 2, 4]
    selection_sort(list)
    print(list)

    list = [3, 9, 2, 1]
    selection_sort(list)
    print(list)


main()

File: sort/selection_sort.js
function selectionSort(arr) {
  for (let i = 0; i < arr.length - 1; i++) {
    let minIndex = i;
    for (let curIdx = i + 1; curIdx < arr.length; curIdx++) {
      if (arr[curIdx] < arr[minIndex]) {
        minIndex = curIdx;
      }
    }
    if (minIndex != i) {
      let temp = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = temp;
    }
  }
  return arr;
}

console.log(selectionSort([29, 10, 14, 37, 13]));
